src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java

```diff
@@ -0,0 +1,260 @@
+import com.sonar.orchestrator.build.MavenBuild;
+import com.sonar.orchestrator.container.Edition;
+import com.sonar.orchestrator.container.Server;
+import com.sonar.orchestrator.http.HttpMethod;
+import com.sonar.orchestrator.http.HttpResponse;
+import com.sonar.orchestrator.locator.FileLocation;
+import com.sonar.orchestrator.locator.MavenLocation;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.assertj.core.api.Assertions;
+import org.fest.util.Collections;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.fail;
+
+/**
+ * a lot of code from https://github.com/SonarSource/sonar-java/blob/5aaaeba90cd85a40391706ad2f806d6a156f80cb/its/ruling/src/test/java/org/sonar/java/it/JavaRulingTest.java
+ * and from
+ * https://github.com/SonarSource/sonar-jacoco/blob/master/its/src/test/java/org/sonar/plugins/jacoco/its/JacocoTest.java
+ */
+public class RunPluginTest {
+    private static final Logger LOG = LoggerFactory.getLogger(RunPluginTest.class);
+    private static final int LOGS_NUMBER_LINES = 200;
+    private static final String TRUE = "true";
+    private static final String PROJECT_KEY = "com.puppycrows.tools:checkstyle";
+    private static final String PROJECT_NAME = "integration-test-project";
+    private static final List<String> DEACTIVATED_RULES = Collections.list(
+            "com.puppycrawl.tools.checkstyle.checks.coding.MissingCtorCheck",
+            "com.puppycrawl.tools.checkstyle.checks.design.DesignForExtensionCheck",
+            "com.puppycrawl.tools.checkstyle.checks.imports.ImportControlCheck",
+            "com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocPackageCheck",
+            "com.puppycrawl.tools.checkstyle.checks.javadoc.WriteTagCheck",
+            "com.puppycrawl.tools.checkstyle.checks.UncommentedMainCheck");
+
+    private static Orchestrator orchestrator;
+
+    @Rule
+    public final TemporaryFolder temp = new TemporaryFolder();
+
+    @BeforeClass
+    public static void beforeAll() {
+        orchestrator = Orchestrator.builderEnv()
+                .setZipFile(new File("target/temp-downloads/sonar-application-7.9.2.zip"))
+                //.setSonarVersion(System.getProperty("sonar.runtimeVersion", "7.9.2"))
+                .setEdition(Edition.COMMUNITY)
+                .addPlugin(FileLocation.byWildcardMavenFilename(new File("target"),
+                        "checkstyle-sonar-plugin-*.jar"))
+                .addPlugin(MavenLocation.of("org.sonarsource.sonar-lits-plugin",
+                        "sonar-lits-plugin",
```

Comment:
why we need this plugin ?

Reply:
i actually didn't find much documentation about it but from playing around with the test project and all rules activated it actually finds out the issues found in the project, if none found the file is empty and, hence, the build passes.
i thought in the end it makes sense to make sure the test project actually passes all activated rules.

Reply:
ok, let it be for some time here , but better to not test not our plugins.